### DISCUSSION TITLE
Thoughts about new CLI (WIP ; will PR soon)

### DISCUSSION THREAD

**Discussion by p0p4k** [coqui-ai/TTS#1685] [Ideas]
I found TTS command line interface to be **slightly**  _un-intuitive_ ; be it typing complete model names OR re-loading models for every "--text" run even on the same model. On top of it, typing `--"parameter" "value"` seems so outdated. So, just to spice some things up, introducing a better and _modern_ CLI. 

![image](https://user-images.githubusercontent.com/8834712/175345990-d0d3ed4d-9c2e-4603-bb92-0ce4e62b9efa.png)

- It is a WIP.
- Based on [python-inquirer](https://github.com/magmax/python-inquirer) package.
- My plan is to add all the features from existing CLI (with the help of synthesize.py).
- Supports holding the loaded model in memory to avoid re-loading it again unless "tts restarted".
- Supports changing speaker_idx and other multi-speaker params for the same loaded model without re-loading it.
- Gonna update the description as I add more features.
- Please comment if you have any new features in your mind.
![image](https://user-images.githubusercontent.com/8834712/175347141-30830386-1376-4c25-b99a-b6981c0b13f5.png)

**Comment by lexkoro** [comment]
Will running the CLI in a bash script still be possible?

**Reply by p0p4k** [reply]
What does running in a bash script actually mean? I am confused. 
Right now, you can type "tts-cli" in cmd/shell and run the interface. If you write something like "python inquire.py" in the script.sh , it will run the interface as well. But, then do you want to automate the inputs, for testing and such? Is that the intent? Thanks.

**Reply by lexkoro** [reply]
Some people have written bash scripts where they process large portions of text in a bash script and synthesize each sentence separately for example using the tts command in a loop.

Example: https://github.com/coqui-ai/TTS/discussions/1622

Also yeah sorry, the question was formulated kinda unclear.

**Reply by p0p4k** [reply]
@lexkoro Hope you had a good weekend.
After giving it some thought, I think in that case we keep both 'tts' (existing cli) and 'tts-cli' (this module), 
'tts' can be used for auto-scripts, tests, etc., while 'tts-cli' might be useful for complete beginners who would want an easier graphical experience (?).
Either way, it is still in early stages and if I can find a way to integrate 'tts-cli' for bash scripts, I will update here. Thanks for the suggestion.

**Comment by erogol** [comment]
I think instead of selecting something from a long list of models, it is easier to write its ID. 

Something like

``` console
$ Choose a TTS model (type model ID): 4
```

**Reply by p0p4k** [reply]
Here, we are typing nothing, just using arrow keys to choose the model. I can change to `{ID} | {name}` if you suggest.

**Reply by erogol** [reply]
Yes, I know, but when selecting the 50th model, one needs to press down 50 times. But one could write 50 quickly :)

**Reply by p0p4k** [reply]
Although, the down button hard pressed, slides down the list real quick, I did not think it through! 
Makes sense for longer lists to just type the ID.
Slightly busy with other things this week, will try to finish this within 2-3 weeks and request review. Thanks for the feedback, as usual! 🙌
